src/plugins/data/public/plugin.ts

```diff
@@ -210,6 +210,14 @@ export class DataPublicPlugin
       storage: this.storage,
     });
 
+    /*
+      Clear any open sessions upon navigation to make sure they are not used mistakenly by
+      another application.
+     */
+    core.application.currentAppId$.subscribe(() => {
```

Comment:
This concerns me. I'd not do it in data plugin for the same reasons why we are not managing URL from a data plugin (filters, timeRange, refreshInterval).

I think instead apps should explicitly handle it.
Maybe we could introduce a `dev` warning, to check here if session wasn't cleaned on app unmount. 
But I am not sure that we should `clear()` here ourselves.

Reply:
One more question here ^^ 
@lizozom, what do you sync? I think discover should handle this when navigating outside of discover?